### PR TITLE
'all platforms' now includes Android (bug 1121653)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1121653](https://bugzilla.mozilla.org/show_bug.cgi?id=1121653)

This follows #395

### CODE REVIEW

**Comment by magopian** [comment]
@kmaglione @clouserw r?

**Comment by magopian** [comment]
I need to fix all those failing tests, please wait a bit more before reviewing ;)

**Comment by kmaglione** [comment]
Looks good aside from the failing tests.

**Comment by magopian** [comment]
Tests are now passing, @kmaglione could you please have another look? I did change a few things, especially https://github.com/mozilla/olympia/pull/433/files#diff-21fff690d768673e64a18d2fd3685d45L247 (i'm pretty sure this piece of code isn't very useful anymore, since the `MOBILE` app should be deprecated).

**Comment by kmaglione** [comment]
Sorry, I started reviewing this last week, but got hung up looking at other uses of `amo.MOBILE` elsewhere in the codebase that need to be fixed. I think I'll ignore them for now and file a separate bug.

File: apps/versions/models.py [hunk 1]
```diff
@@ -244,8 +244,8 @@ def compatible_platforms(self):
         The result is based on which app(s) the version targets.
         """
         app_ids = [a.application for a in self.apps.all()]
-        targets_mobile = amo.MOBILE.id in app_ids
-        targets_other = any((id_ != amo.MOBILE.id) for id_ in app_ids)
+        targets_mobile = amo.ANDROID.id in app_ids
```
Comment (kmaglione) [inline apps/versions/models.py]: It's interesting that this didn't cause problems before... From what I can tell, it should have only made it possible to upload files for Android Firefox if they also claimed to support XUL Fennec, or specified a platform in the XPI.

**Comment by kmaglione** [comment]
The code changes look good, but the client-side submission flow code makes me itch, so I want to QA this myself before it goes to prod.

r+

**Comment by magopian** [comment]
Thanks a lot for the review, please let me know when you believe this is good to merge.

**Comment by magopian** [comment]
Also, just fyi, i've file https://bugzilla.mozilla.org/show_bug.cgi?id=1126847 a few days ago

**Comment by kmaglione** [comment]
Please merge now. I'll test it on dev.

**Pull request merged**